src/lib.rs

```diff
@@ -64,6 +64,21 @@ fn __init(w: &mut impl std::io::Write) -> std::io::Result<()> {
     // Read the git revision from the git repository containing the code being
     // built.
     if git_sha.is_none() {
+        // Require the build script to rerun if relavent git state changes which
```

Comment:
Note that these lines are only present with importing stellar-xdr from a git repo. When imported from crates.io they will never be triggered because there's no point since there will be no .git dir.